### ISSUE TITLE
Exploring Numerical Data

### ISSUE THREAD

**Issue by DevCEDTeam** [DevCEDTeam/CED#195]
[Exploring Numerical Data](https://chatgpt.com/s/t_68dee8d84820819190344f754cfd9cd7)

**Comment by DevCEDTeam** [comment]
Excellent — here’s the **enhanced, Chicago-formatted version** of your structured summary with **sample Python scripts** appended for each section.
All code examples are concise, ready to run in Jupyter, and written for clarity and reproducibility.

---

# **I. Exploring Numerical Data (with Python Examples)**

---

## **1. Start with the Basics**

Before performing visualizations or modeling, first examine the dataset’s structure and completeness.

### **a. Key Steps**

* Determine dataset **shape** (rows × columns).
* Inspect **data types** (integer, float, string).
* Check for **missing values** or incomplete records.

### **b. Sample Python Script**

```python
import pandas as pd

# Load dataset
df = pd.read_csv("data.csv")

# Shape and data type overview
print("Shape:", df.shape)
print("\nData Types:\n", df.dtypes)

# Check for missing values
print("\nMissing Values:\n", df.isnull().sum())
```

---

## **2. Descriptive Statistics**

Use simple summary metrics to identify patterns and detect anomalies.

### **a. Core Metrics**

* **Mean, Median, Mode:** Central tendency indicators
* **Standard Deviation, Variance:** Measure spread
* **Min, Max, Percentiles:** Highlight range and extreme values

### **b. Sample Python Script**

```python
# Basic descriptive statistics
print(df.describe())

# Mode for categorical/numerical columns
print("\nMode:\n", df.mode().iloc[0])
```

---

## **3. Visualize the Distribution**

Visualize the spread and shape of numerical data to detect patterns, skewness, or multimodal behavior.

### **a. Visualization Techniques**

* **Histograms:** Frequency distribution
* **Boxplots:** Outlier detection and spread
* **Density Plots:** Smooth probability density view

### **b. Sample Python Script**

```python
import matplotlib.pyplot as plt
import seaborn as sns

# Histogram
sns.histplot(df['variable_name'], bins=20, kde=False)
plt.title('Histogram of Variable')
plt.show()

# Boxplot
sns.boxplot(x=df['variable_name'])
plt.title('Boxplot of Variable')
plt.show()

# Density plot
sns.kdeplot(df['variable_name'], fill=True)
plt.title('Density Plot of Variable')
plt.show()
```

---

## **4. Look for Outliers and Anomalies**

Outliers can be statistical noise or valuable signals.

### **a. Techniques**

* **Z-score Method:** Based on standard deviation distance from mean.
* **IQR Method:** Uses the interquartile range to identify extreme points.

### **b. Sample Python Script**

```python
import numpy as np
from scipy import stats

# Z-score method
z_scores = np.abs(stats.zscore(df['variable_name']))
outliers_z = df[z_scores > 3]

# IQR method
Q1 = df['variable_name'].quantile(0.25)
Q3 = df['variable_name'].quantile(0.75)
IQR = Q3 - Q1
outliers_iqr = df[(df['variable_name'] < (Q1 - 1.5 * IQR)) | 
                  (df['variable_name'] > (Q3 + 1.5 * IQR))]

print("Outliers (Z-score):", outliers_z.shape[0])
print("Outliers (IQR):", outliers_iqr.shape[0])
```

---

## **5. Think About Scale**

Scaling ensures consistent variable influence in machine learning algorithms.

### **a. Normalization vs. Standardization**

* **Normalization (Min–Max Scaling):** Scales features between 0 and 1.
* **Standardization (Z-score Scaling):** Centers data around mean 0 and variance 1.

### **b. Sample Python Script**

```python
from sklearn.preprocessing import MinMaxScaler, StandardScaler

# Normalization
scaler_minmax = MinMaxScaler()
df_normalized = scaler_minmax.fit_transform(df[['feature1', 'feature2']])

# Standardization
scaler_standard = StandardScaler()
df_standardized = scaler_standard.fit_transform(df[['feature1', 'feature2']])
```

---

## **6. Explore Relationships**

Once variables are individually understood, explore their interrelationships.

### **a. Methods**

* **Scatter Plots:** Visualize two-variable linearity
* **Heatmaps:** Show correlation strengths
* **Pairplots:** Compactly display all pairwise variable relations

### **b. Sample Python Script**

```python
# Scatter plot
sns.scatterplot(x='feature1', y='feature2', data=df)
plt.title('Scatter Plot: Feature1 vs Feature2')
plt.show()

# Correlation heatmap
corr = df.corr()
sns.heatmap(corr, annot=True, cmap='coolwarm')
plt.title('Correlation Heatmap')
plt.show()

# Pairplot
sns.pairplot(df)
plt.suptitle('Pairplot of All Features', y=1.02)
plt.show()
```

---

## **7. Core Mathematical Formulas**

For analytical rigor, recall the essential statistical equations:

### **a. Variance**

[
\sigma^2 = \frac{\sum_i (x_i - \bar{x})^2}{N}
]

### **b. Standard Deviation**

[
\sigma = \sqrt{\frac{\sum_i (x_i - \bar{x})^2}{N}}
]

### **c. Sample Computation**

```python
# Variance and Standard Deviation
variance = df['variable_name'].var()
std_dev = df['variable_name'].std()
print(f"Variance: {variance:.3f}, Standard Deviation: {std_dev:.3f}")
```

---

## **II. Gilbert Technique Summary**

1. **Comprehend** — begin with structure (shape, type, gaps).
2. **Quantify** — describe statistically (mean, median, spread).
3. **Visualize** — reveal trends and anomalies graphically.
4. **Normalize** — align scales for meaningful comparison.
5. **Relate** — explore interactions across variables.
6. **Refine** — isolate outliers and patterns for interpretation.

This systematic order enhances clarity, reduces analytical error, and strengthens data-driven inference.

---

Would you like me to **export this as a formatted academic PDF handout** with each Python code block highlighted and labeled under section headers (ideal for classroom or workshop use)?
